### DISCUSSION TITLE
Set temp directory independent of output directory

### DISCUSSION THREAD

**Discussion by MacSass** [mifi/lossless-cut#2112] [Ideas]
Hi,
I would like to keep the output of my files in the same folder as the original files - which can be set easily in lossless-cut.

Now for performance reasons I would like to set the temporary directory to a different place than the output, as my final files are on an HD and having temp files created there slow process down due to HD being slow when reading/writing several files at the same time.

Thus I wan the temp directory on my SSD (and temporary files cleaned after final export) and the output file in the same place as the original file. I don´t say a way to do that currently.

I would propose to make setting for temporary directory independent from output, as I see those as two different things.

Thanks - MacSass

**Comment by mifi** [comment]
Hi. I believe this is the same as #1536

**Reply by MacSass** [reply]
Not 100% sure - but yes - for sure it is related.
My concern is that I want to have the temp files on a "temp" folder that sits on my SSD, while the source and destination file (I merge cuts) should be on my HD download folder.
This is mainly to speed up the working as a HD gets very slow when it is read and written to at the same time.
I did not look much in the "save project file", as I auto delete that (as I do with the temp working files).

So - yes that might be related, not sure it is the same though.
Thanks for your response!

**Reply by mifi** [reply]
ah. by "temp" files do you mean the intermediate files like all exported segments files when doing cut+merge (that later get merged together into one file and then the individual files get deleted)?

I think most people want to have the temp files in the same directory as the output because the temp files can sometimes take a lot of space and they don't always have so much space on their SSD. i added your suggestion here #2115

**Reply by MacSass** [reply]
Yes - that is correct. As they are only "temp" files there should usually be enough space on the SSD - if you SSD does not have a couple of GB left that is causing other issues.
As said - having those files on the HD, where they are read and merged into one single file is awfully slow on a HD.
Looking at my previous "cut&merge" solution, called "VidCutter" ( https://github.com/ozmartian/vidcutter)  it is way faster in the final processing of the files, as it keeps the temp files in the system temp directory.
Anyhow - your solution overall has a much better UI and obviously better maintenance and support :-) ... so thank you for taking that into account ...
Regards - MacSass
PS: Final note: For some reason "Vidcutter" also is a faster in "scrolling/preview" - not sure how they do it, but something I noticed.
Still for many other PROs for your solution I plan to leave vidcutter behind.

**Reply by mifi** [reply]
alright, thanks for the useful input!

> PS: Final note: For some reason "Vidcutter" also is a faster in "scrolling/preview"

do you mean faster scrubbing on the timeline, so that the preview image follows the cursor more closely? i actually improved that very recently in 4ccf3180caae90beee07742ba7e0bd3cadd45722 - you can try the latest nightly build if you want to test it

**Reply by MacSass** [reply]
In regard to your final note: Just tested the nightly and wow! I´m impressed. Starts faster and scrolling in timeline is way more faster too. 
It is now up to par with my previous solution (VidCutter) - very nice.
As I guess the export speed will improve as soon as I have the temp files on SSD, I do not see anymore reason not to switch.
You app has the better GUI, more features and good development and support ... I will try changing to it and see how it goes (I might have some more inputs in the future :-) ).
Thanks for your great job!
